lib/perl/Genome/Model/Tools/Picard.pm

```diff
@@ -318,6 +339,11 @@ sub run_java_vm {
 sub create {
     my $class = shift;
     my $self = $class->SUPER::create(@_);
+
+    if (not defined $self->use_version) {
+        die $self->error_message('Cannot pass undef to Picard::use_version');
```

Comment:
I'm worried that this will actually cause crashes in all sorts of places that aren't picked up by the test suite.

The test failures for earlier commits in this PR were due to the Bwamem test not setting any value for `picard_version` on the test `AlignmentResult` object. Then, `AlignmentResult::Bwamem` would blindly pass the undefined `$self->picard_version` to `Picard::use_version`. In other words, `use_version` would be set to `undef`, not the default value.

However, until now, `use_version` was only used like so:
```
sub picard_path {
    my $self = shift;
    return $self->path_for_picard_version($self->use_version); # in this case, undef
}

sub path_for_picard_version {
    my ($class, $version) = @_;
    $version ||= $PICARD_DEFAULT; # the default gets filled in
    ...
}
```

I think the reasoning behind this was to allow someone to call `GMT::Picard->path_for_picard_version()` and get the path to the "default" Picard. However this had the side effect of masking bad behavior from callers passing in an undefined `use_version`. Until now...

I'm hesitant to push code that could break poorly written callers but I also don't want to further legitimize the use of default version (by filling in the default if someone incorrectly tries to use `undef` as a version). In my opinion there should be no default version. Either pick one or always use the latest version if none is specified.